diff --git a/types/build.gradle.kts b/types/build.gradle.kts
index 665e94e..37bdfae 100644
--- a/types/build.gradle.kts
+++ b/types/build.gradle.kts
@@ -0,0 +1,2 @@
+import java.util.Locale
+
@@ -3,0 +6 @@ plugins {
+    `maven-publish`
@@ -31,0 +35,12 @@ java {
+
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
