diff --git a/types/build.gradle.kts b/types/build.gradle.kts
index 7d4ed6e..7c56148 100644
--- a/types/build.gradle.kts
+++ b/types/build.gradle.kts
@@ -16,9 +16,12 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+import java.util.Locale
+
 plugins {
     kotlin("jvm") version "1.9.22"
     kotlin("plugin.serialization") version "1.9.22"
+    `maven-publish`
 }
 
 group = "gg.skytils.hypixel.types"
@@ -46,4 +49,16 @@ dependencies {
 
 java {
     toolchain.languageVersion = JavaLanguageVersion.of(8)
-}
\ No newline at end of file
+}
+
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
