diff --git a/build.gradle.kts b/build.gradle.kts
index fed6ddee..f2168a19 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -20,0 +21 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.util.Locale
@@ -34,0 +36 @@ plugins {
+    `maven-publish`
@@ -281,0 +284,12 @@ signing {
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
+
diff --git a/events/build.gradle.kts b/events/build.gradle.kts
index 9e967282..64f45149 100644
--- a/events/build.gradle.kts
+++ b/events/build.gradle.kts
@@ -18,0 +19 @@ import org.apache.tools.ant.filters.FixCrLfFilter
+import java.util.Locale
@@ -23,0 +25 @@ plugins {
+    `maven-publish`
@@ -50,0 +53,12 @@ tasks.withType<AbstractArchiveTask> {
+
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
