diff --git a/build.gradle.kts b/build.gradle.kts
index fed6ddee..f2168a19 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -18,6 +18,7 @@
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 import net.fabricmc.loom.task.RemapJarTask
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.util.Locale
 import java.security.MessageDigest
 
 plugins {
@@ -32,6 +33,7 @@ plugins {
     id("org.jetbrains.gradle.plugin.idea-ext") version "1.1.7"
 
     signing
+    `maven-publish`
 }
 
 version = "1.9.0-pre3"
@@ -279,6 +281,18 @@ signing {
     }
 }
 
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
+
 /**
  * Builds the dependency notation for the named Ktor [module] at the given [version].
  *
diff --git a/events/build.gradle.kts b/events/build.gradle.kts
index 9e967282..64f45149 100644
--- a/events/build.gradle.kts
+++ b/events/build.gradle.kts
@@ -16,11 +16,13 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 import org.apache.tools.ant.filters.FixCrLfFilter
+import java.util.Locale
 
 plugins {
     kotlin("jvm") version "1.9.22"
     id("gg.essential.loom") version "1.3.12"
     id("gg.essential.defaults") version "0.3.0"
+    `maven-publish`
 }
 
 dependencies {
@@ -47,4 +49,16 @@ tasks.processResources {
 tasks.withType<AbstractArchiveTask> {
     isPreserveFileTimestamps = false
     isReproducibleFileOrder = true
-}
\ No newline at end of file
+}
+
+publishing {
+    publications {
+        register("mavenJava", MavenPublication::class) {
+            groupId = project.group.toString()
+            artifactId = project.name.lowercase(Locale.ROOT)
+            version = project.version.toString()
+
+            from(components["java"])
+        }
+    }
+}
